gamemodes/terrortown/gamemode/shared/sh_player_ext.lua

```diff
@@ -287,22 +287,75 @@ function plymeta:CanCarryWeapon(wep)
 	if not wep or not wep.Kind then
 		return false
 	end
-
+	
+	--appeareantly TTT can't handle two times the same weapon
+	for k, v in pairs(self:GetWeapons()) do
+		if IsValid(wep) and IsValid(v) and v:GetClass() == wep:GetClass() then
+			return false
+		elseif istable(wep) and IsValid(v) and v:GetClass() == wep.ClassName then
+			return false
+		end
+	end
+	
 	return self:CanCarryType(wep.Kind)
 end
 
 function plymeta:CanCarryType(t)
 	if not t then
 		return false
 	end
+	
+	return InventorySlotFree(self, t)
+end
 
-	for _, w in pairs(self:GetWeapons()) do
-		if w.Kind and w.Kind == t then
-			return false
+function plymeta:GetInventory()
+	CleanupInventoryIfDirty(self)
```

Comment:
I suggest to clean direclty after modifying the data, not on trying to access it. This could cause issues.

Reply:
The idea behind that is a kind of caching functionality, which only updates the inventory if needed and dirty (e.g. GetInventory is called). Updating weapons work great on Server side, but doesn't on Client, because Entities are updated late/are invalid for some reason.

A solution to beautify the code would be to always use caching (not only when something is invalid). However, this would probably be less runtime efficient.

(the same comment but now in the right place ^^)

Reply:
So I noticed, that when a player buys a weapon from Shop the WeaponEquip hook is called on SERVER. There the entity is valid. If i send a message from here to the Client with the same entity, it is a NULL Entity. 

Yeah of cause that would be perfect, but leads to weird behaviour on client as described above. But the cache is only updated once (if weapons were changed), because after that the cache isn't dirty anymore. This "Dirty-Flag" is an often used technique.